pkg/apis/ndm/v1alpha1/types.go

```diff
@@ -71,3 +71,37 @@ type StoragePoolList struct {
 
 	Items []StoragePool `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=diskobject
+
+// DiskObject describes disk resource.
+type DiskObject struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+
+	Spec DiskObjectSpec `json:"spec"`
+}
+
+// DiskObjectSpec is the specification for the disk stored as CRD
+type DiskObjectSpec struct {
+	Uuid       string `json:"uuid"`  //unique identifier for the disk
+	Name       string `json:"name"`  //disk name (e.g. sdb, sdc)
+	Path       string `json:"path"`  //disk path (e.g. /dev/sdb)
+	Size       string `json:"size"`  //disk size (e.g. 200G)
```

Comment:
Its better to put this number in bytes such that UI's and what not, don't have to parse our strings and can decide how they want to display the numbers themselves.